cardano-node/src/Cardano/Node/Configuration/Logging.hs

```diff
@@ -211,3 +240,51 @@ createLoggingLayer ver nodeConfig' = do
 
 shutdownLoggingLayer :: LoggingLayer -> IO ()
 shutdownLoggingLayer = shutdown . llSwitchboard
+
+-- The node provides the basic node's information for TraceForwarderBK.
+-- It will be sent once TraceForwarderBK is connected to an external process
+-- (for example, RTView).
+nodeBasicInfo :: NodeConfiguration
+              -> Consensus.Protocol IO blk (BlockProtocol blk)
+              -> UTCTime
+              -> IO [LogObject Text]
+nodeBasicInfo nc p nodeStartTime' = do
+  meta <- mkLOMeta Notice Public
+  let cfg = pInfoConfig $ Consensus.protocolInfo p
+      protocolDependentItems =
+        case p of
+          Consensus.ProtocolByron {} ->
+            let DegenLedgerConfig cfgByron = Consensus.configLedger cfg
+            in getGenesisValuesByron cfg cfgByron
+          Consensus.ProtocolShelley {} ->
+            let DegenLedgerConfig cfgShelley = Consensus.configLedger cfg
+            in getGenesisValues "Shelley" cfgShelley
+          Consensus.ProtocolCardano {} ->
+            let CardanoLedgerConfig cfgByron cfgShelley cfgAllegra cfgMary = Consensus.configLedger cfg
+            in getGenesisValuesByron cfg cfgByron
+               ++ getGenesisValues "Shelley" cfgShelley
+               ++ getGenesisValues "Allegra" cfgAllegra
+               ++ getGenesisValues "Mary"    cfgMary
+      items = nub $
+        [ ("protocol",      pack . protocolName $ ncProtocol nc)
+        , ("version",       pack . showVersion $ version)
+        , ("commit",        gitRev)
+        , ("nodeStartTime", show nodeStartTime')
+        ] ++ protocolDependentItems
+      logObjects =
+        map (\(nm, msg) -> LogObject ("basicInfo." <> nm) meta (LogMessage msg)) items
+  return logObjects
+ where
+  getGenesisValuesByron cfg config =
+    let genesis = byronLedgerConfig config
+    in [ ("systemStartTime",  show (WCT.getSystemStart . getSystemStart $ Consensus.configBlock cfg))
+       , ("slotLengthByron",  show (WCT.getSlotLength . fromByronSlotLength $ genesisSlotLength genesis))
+       , ("epochLengthByron", show (unEpochSize . fromByronEpochSlots $ Gen.configEpochSlots genesis))
+       ]
+  getGenesisValues era config =
+    let genesis = shelleyLedgerGenesis $ shelleyLedgerConfig config
+    in [ ("systemStartTime",          show (SL.sgSystemStart genesis))
+       , ("slotLength" <> era,        show (WCT.getSlotLength . WCT.mkSlotLength $ SL.sgSlotLength genesis))
+       , ("epochLength" <> era,       show (unEpochSize . SL.sgEpochLength $ genesis))
+       , ("slotsPerKESPeriod" <> era, show (SL.sgSlotsPerKESPeriod genesis))
+       ]
```

Comment:
Is it appropriate to just use `show` on these values? What does the output look like?

I'd much prefer `render*` functions that ensure these values are appropriately formatted and rendered to `Text`.

Reply:
> If the text representation of the values in nodeBasicInfo looks okay, then we can merge this.

@intricate Yes, the representation of these values is okay: it's the simplest possible. :-) For example, `slotLength` will be rendered not as:

```
SlotLength { getSlotLength = 1 }
```

but as:

```
1
```